src/tasks.rs

```diff
@@ -58,18 +59,63 @@ impl fmt::Debug for Task {
 }
 
 impl Task {
-    pub fn run<DB: WriteResults>(&self, ex: &Experiment, db: &DB) -> Result<()> {
+    pub fn needs_exec<DB: WriteResults>(&self, ex: &Experiment, db: &DB) -> bool {
+        // A prepare step should already be executed, and other steps only if were not executed
```

Comment:
Should 'already' mean 'always' here? Otherwise I'm a bit confused about how to reconcile the comment with code that seems to suggest a prepare step is always ready to go.